### PR TITLE
[BugFix] Fix null pointer error of column shared_ptr in CompactionState::_do_load.

### LINKED ISSUE #11001: [Bug] core when CompactionState::_do_load.
### Real behavior (Required)
```
Core was generated by `/opt/starrocks/be/lib/starrocks_be'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release (this=0x51167c0 <vtable for starrocks::ZoneMapPB+16>) at /usr/include/c++/10.3.0/ext/atomicity.h:84
84      /usr/include/c++/10.3.0/ext/atomicity.h: No such file or directory.
[Current thread is 1 (LWP 91766)]
(gdb) set substitute-path '/root/starrocks' '/home/idpservice/starrocks-2.1.11'
(gdb) bt
#0  std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release (this=0x51167c0 <vtable for starrocks::ZoneMapPB+16>) at /usr/include/c++/10.3.0/ext/atomicity.h:84
#1  std::__shared_count<(__gnu_cxx::_Lock_policy)2>::~__shared_count (this=0x39a90e2a0, __in_chrg=<optimized out>) at /usr/include/c++/10.3.0/bits/shared_ptr_base.h:733
#2  std::__shared_ptr<starrocks::vectorized::Column, (__gnu_cxx::_Lock_policy)2>::~__shared_ptr (this=0x39a90e298, __in_chrg=<optimized out>) at /usr/include/c++/10.3.0/bits/shared_ptr_base.h:1183
#3  std::shared_ptr<starrocks::vectorized::Column>::~shared_ptr (this=0x39a90e298, __in_chrg=<optimized out>) at /usr/include/c++/10.3.0/bits/shared_ptr.h:121
#4  starrocks::vectorized::CompactionSemgentState::~CompactionSemgentState (this=0x39a90e298, __in_chrg=<optimized out>) at /root/starrocks/be/src/storage/update_compaction_state.h:19
#5  std::_Destroy<starrocks::vectorized::CompactionSemgentState> (__pointer=0x39a90e298) at /usr/include/c++/10.3.0/bits/stl_construct.h:140
#6  std::_Destroy_aux<false>::__destroy<starrocks::vectorized::CompactionSemgentState*> (__last=<optimized out>, __first=0x39a90e298) at /usr/include/c++/10.3.0/bits/stl_construct.h:152
#7  std::_Destroy<starrocks::vectorized::CompactionSemgentState*> (__last=<optimized out>, __first=<optimized out>) at /usr/include/c++/10.3.0/bits/stl_construct.h:185
#8  std::_Destroy<starrocks::vectorized::CompactionSemgentState*, starrocks::vectorized::CompactionSemgentState> (__last=<optimized out>, __first=<optimized out>) at /usr/include/c++/10.3.0/bits/alloc_traits.h:738
#9  std::vector<starrocks::vectorized::CompactionSemgentState, std::allocator<starrocks::vectorized::CompactionSemgentState> >::_M_erase_at_end (__pos=0x39a90e298, this=<error reading variable: Cannot access memory at address 0x7fad4220b700>)
    at /usr/include/c++/10.3.0/bits/stl_vector.h:1796
#10 std::vector<starrocks::vectorized::CompactionSemgentState, std::allocator<starrocks::vectorized::CompactionSemgentState> >::resize (__new_size=<optimized out>, this=<error reading variable: Cannot access memory at address 0x7fad4220b700>)
    at /usr/include/c++/10.3.0/bits/stl_vector.h:942
#11 starrocks::vectorized::CompactionState::_do_load (this=<error reading variable: Cannot access memory at address 0x7fad4220b700>, rowset=0xfa64c51e0) at /root/starrocks/be/src/storage/update_compaction_state.cpp:46
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.1.11

### PR DESCRIPTION
**Pull request by Linkerist**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11001

## Problem Summary(Required) ：

Jump out of current for loop if segment iterator is nullptr. Otherwise, the columns will throw a null pointer error elsewhere.

### CODE REVIEW

**Comment by Linkerist** [comment]
run starrocks_be_unittest

File: be/src/storage/update_compaction_state.cpp [hunk 1]
```diff
@@ -74,27 +74,26 @@ Status CompactionState::_do_load(Rowset* rowset) {
     auto chunk = chunk_shared_ptr.get();
 
     for (size_t i = 0; i < itrs.size(); i++) {
-        auto itr = itrs[i].get();
```
Comment (trueeyu) [inline be/src/storage/update_compaction_state.cpp:77]: There's not much difference between the two algorithms ？

**Review by trueeyu — DISMISSED** [review]

**Comment by trueeyu** [comment]
Still don't understand why it crashes

**Review by trueeyu — COMMENTED** [review]
May be caused by other reason

**Review by chaoyli — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by decster** [comment]
run starrocks_admit_test

**Comment by decster** [comment]
run starrocks_admit_test

**Comment by Linkerist** [comment]
run starrocks_admit_test

**Comment by Linkerist** [comment]
run starrocks_admit_test

**Comment by Linkerist** [comment]
run starrocks_admit_test

**Pull request merged**